### ISSUE TITLE
Import error in Ubuntu 22.04 LTS

### ISSUE THREAD

**Issue by grassFlamingo** [tensorly/tensorly#473]
Dear developers,

When I try import tensorly (0.8.0, 7a252eb, installed via `make install`) in Ubuntu 22.04 LTS with Python 3.7 by
```python
import tensorly
```
it will raise an error
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/xxx/tensorly/tensorly/__init__.py", line 10, in <module>
    from .cp_tensor import (
  File "/xxx/tensorly/tensorly/cp_tensor.py", line 7, in <module>
    from ._factorized_tensor import FactorizedTensor
  File "/xxx/tensorly/tensorly/_factorized_tensor.py", line 2, in <module>
    from . import tenalg
  File "/xxx/tensorly/tensorly/tenalg/__init__.py", line 91, in <module>
    TenalgBackendManager.initialize_backend()
  File "/xxx/tensorly/tensorly/backend/__init__.py", line 291, in initialize_backend
    cls.set_backend(backend_name)
  File "/xxx/tensorly/tensorly/backend/__init__.py", line 335, in set_backend
    backend = cls.load_backend(backend)
  File "/xxx/tensorly/tensorly/tenalg/__init__.py", line 81, in load_backend
    importlib.import_module(f"tensorly.tenalg.{backend_name}_tenalg")
  File "/xxx/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/xxx/tensorly/tensorly/tenalg/core_tenalg/__init__.py", line 4, in <module>
    from .generalised_inner_product import inner
  File "<fstring>", line 1
    (tensor1.shape=)
                  ^
SyntaxError: invalid syntax
```

While checking the codes, it seems that some f-strings are incorrect. They looks like
```python
f"{variable =}"
```
I think that the equal sign is not necessary. Therefore, I have made the following changes.
```
diff --git a/tensorly/tenalg/core_tenalg/generalised_inner_product.py b/tensorly/tenalg/core_tenalg/generalised_inner_product.py
index 7544303..0e1e338 100644
--- a/tensorly/tenalg/core_tenalg/generalised_inner_product.py
+++ b/tensorly/tenalg/core_tenalg/generalised_inner_product.py
@@ -31,7 +31,7 @@ def inner(tensor1, tensor2, n_modes=None):
                 "Taking a generalised product between two tensors without specifying common modes"
                 " is equivalent to taking inner product."
                 "This requires tensor1.shape == tensor2.shape."
-                f"However, got {tensor1.shape=} and {tensor2.shape=}"
+                f"However, got {tensor1.shape} and {tensor2.shape}"
             )
         return T.sum(tensor1 * tensor2)
 
@@ -45,7 +45,7 @@ def inner(tensor1, tensor2, n_modes=None):
     if common_modes != shape_t2[:n_modes]:
         raise ValueError(
             f"Incorrect shapes for inner product along {n_modes} common modes."
-            f"{tensor_1.shape=}, {tensor_2.shape=}"
+            f"{tensor_1.shape}, {tensor_2.shape}"
         )
     inner_product = T.dot(
         T.reshape(tensor1, (-1, common_size)), T.reshape(tensor2, (common_size, -1))
diff --git a/tensorly/tt_tensor.py b/tensorly/tt_tensor.py
index 1eba071..9787dd0 100644
--- a/tensorly/tt_tensor.py
+++ b/tensorly/tt_tensor.py
@@ -257,7 +257,7 @@ def validate_tt_rank(
         if isinstance(rank, int):
             rank = [1] + [rank] * (n_dim - 1) + [1]
         elif n_dim + 1 != len(rank):
-            message = f"Provided incorrect number of ranks. Should verify len(rank) == tl.ndim(tensor)+1, but {len(rank) = } while tl.ndim(tensor) + 1  = {n_dim+1}"
+            message = f"Provided incorrect number of ranks. Should verify len(rank) == tl.ndim(tensor)+1, but {len(rank)} while tl.ndim(tensor) + 1  = {n_dim+1}"
             raise (ValueError(message))
 
         # Initialization
diff --git a/tensorly/tucker_tensor.py b/tensorly/tucker_tensor.py
index 0f57fb7..82e6a3f 100644
--- a/tensorly/tucker_tensor.py
+++ b/tensorly/tucker_tensor.py
@@ -190,14 +190,14 @@ def tucker_mode_dot(tucker_tensor, matrix_or_vector, mode, keep_dim=False, copy=
         if matrix_or_vector.shape[1] != shape[mode]:
             raise ValueError(
                 f"shapes {shape} and {matrix_or_vector.shape} not aligned in mode-{mode} multiplication: "
-                f"{shape[mode]} ({mode = }) != {matrix_or_vector.shape[1]} (dim 1 of matrix)"
+                f"{shape[mode]} ({mode}) != {matrix_or_vector.shape[1]} (dim 1 of matrix)"
             )
 
     elif tl.ndim(matrix_or_vector) == 1:  # Tensor times vector
         if matrix_or_vector.shape[0] != shape[mode]:
             raise ValueError(
                 f"shapes {shape} and {matrix_or_vector.shape} not aligned for mode-{mode} multiplication: "
-                f"{shape[mode]} ({mode = }) != {matrix_or_vector.shape[0]} (vector size)"
+                f"{shape[mode]} ({mode}) != {matrix_or_vector.shape[0]} (vector size)"
             )
         if not keep_dim:
             contract = True  # Contract over that mode
```

After these, it works on my machine.
Thank you.

**Comment by JeanKossaifi** [comment]
Hi @grassFlamingo - it sounds like you have an older version of Python, the syntax you see was introduced in Python 3.8: https://docs.python.org/3/whatsnew/3.8.html#bpo-36817-whatsnew. For instance, `{shape=}` is a shortcut for `shape={shape}`.

It seems reasonable to me to not use this syntax to support python 3.7 though, could you open a small PR to make the changes?

**Comment by Kuo-TingKai** [comment]
Or I can open a PR to write a 3.7/3.8 compatible version ?

**Comment by JeanKossaifi** [comment]
I've been thinking about this and actually believe it may make more sense to nudge users to update their Python version. We are coming to version 12, I don't think we need to support Python 7.x anymore. I see @yngvem and @MarieRoald also discontinued python 7.x support in tensorly-viz. 

As a side note @Kuo-TingKai, I try never to rely on the system's Python version and instead always install my own, usually via anaconda. I recommend you try this. This has also the advantage of letting you create virtual environments so your main configuration doesn't get messed up.

**Comment by JeanKossaifi** [comment]
@Kuo-TingKai -- if you're still happy to open a small PR to revert those prints that would be great!

**Comment by JeanKossaifi** [comment]
Should be fixed by #482
